.github/workflows/lock.yml

```diff
@@ -0,0 +1,44 @@
+name: 'Lock Inactive Threads'
+
+on:
+  schedule:
+    # midnight UTC, every Wednesday
+    - cron: '0 0 * * 3'
+  # allow manual triggering from GitHub UI
+  workflow_dispatch:
+
+permissions:
+  issues: write
+  pull-requests: write
+
+concurrency:
+  group: lock
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dessant/lock-threads@v4
+        with:
+          github-token: ${{ github.token }}
+          # after how many days of inactivity should a closed issue/PR be locked?
+          issue-inactive-days: '90'
+          pr-inactive-days: '90'
+          # do not close feature request issues...
+          # we close those but track them in https://github.com/microsoft/LightGBM/issues/2302
+          exclude-any-issue-labels: 'feature request'
+          # what labels should be removed prior to locking?
+          remove-issue-labels: 'awaiting response,awaiting review,blocking,in progress'
+          remove-pr-labels: 'awaiting response,awaiting review,blocking,in progress'
```

Comment:
Not sure if this will work with labels with spaces in the names... but I think the only way to test iis to merge to `master`.

Reply:
oh cool idea to use the GitHub CLI! I'd never used that before.